extensions.html

```diff
@@ -182,19 +184,146 @@ <h2>
         <dfn>GamepadHapticActuator</dfn> Interface
       </h2>
       <p>
-        Each <code>GamepadHapticActuator</code> corresponds to a motor or other
-        actuator that can apply a force for the purposes of haptic feedback.
+        A {{GamepadHapticActuator}} corresponds to a configuration of motors or
+        other actuators that can apply a force for the purposes of haptic
+        feedback.
       </p>
       <pre class="idl">
         [Exposed=Window]
         interface GamepadHapticActuator {
           readonly attribute GamepadHapticActuatorType type;
+          boolean canPlayEffectType(GamepadHapticEffectType type);
+          Promise&lt;GamepadHapticsResult&gt; playEffect(
+              GamepadHapticEffectType type,
+              optional GamepadEffectParameters params = {});
           Promise&lt;boolean&gt; pulse(double value, double duration);
+          Promise&lt;GamepadHapticsResult&gt; reset();
         };
       </pre>
+      <p>
+        Instances of {{GamepadHapticActuator}} are created with the internal
+        slots described in the following table:
+      </p>
+      <table class="simple">
+        <tr>
+          <th>
+            Internal slot
+          </th>
+          <th>
+            Initial value
+          </th>
+          <th>
+            Description (non-normative)
+          </th>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for=
+            "GamepadHapticActuator">[[\playingEffectPromise]]</dfn>
```

Comment:
This guard might be better as just a boolean/flag to control the state machine. This is because the actual promise to play something can only be returned once.

However, I'm worried about something... 

Consider the case where you want to do the following:

 1. Lightly "rumble" the right-side of the controller. 
 2. Simultaneously, "rumble" one of the triggers.
 
In theory, it should be possible to activate multiple actuators at the same time, no? 

The current design would prevent this from working:

```JS
const rumble = gamepad.vibrationActuator.play("double-rumble", {intensity: 0.2, duration: 2000});
// reject :( 
const trigger = gamepad.vibrationActuator.play("trigger-rumble", {intensity: 2.0, duration: 2000});  
```

So, I think the guard is actually a Map `map<GamepadHapticActuatorType, boolean>`.  Something like: 

```
GamepadHapticActuator -> "dual-rumble" -> isPlaying ?
GamepadHapticActuator -> "left-trigger" -> isPlaying ?
GamepadHapticActuator -> "right-trigger" -> isPlaying ?
```

Reply:
> In theory, it should be possible to activate multiple actuators at the same time, no?

With the current proposal for trigger rumble, it should work to issue a "trigger-rumble" effect that contains the parameters from the "dual-rumble" effect along with the parameters for the trigger effect. Something like:

```
// rumble and vibrate right trigger
const effectParams = {rightTrigger: 1.0, strongMagnitude: 0.2, weakMagnitude: 0.2, duration: 2000};
gamepad.vibrationActuator.play("trigger-rumble", effectParams);
```

The actuators can be controlled independently by preempting the ongoing effect:

```
// stop trigger vibration, keep rumble
effectParams.rightTrigger = 0.0;
gamepad.vibrationActuator.play("trigger-rumble", effectParams);
```

I think the API should not support simultaneous effects on a single GamepadHapticActuator. If there is an ongoing effect when a new effect is played, the ongoing effect is canceled before the new effect begins.